aospy/automate.py

```diff
@@ -0,0 +1,247 @@
+    """Merge the given dictionaries into single dict.
+
+    Given any number of dicts, shallow copy and merge into a new dict,
+    precedence goes to key value pairs in latter dicts.
+
+    From http://stackoverflow.com/a/26853961/1706640
+    """
+    result = {}
+    for dictionary in dict_args:
+        result.update(dictionary)
+    return result
+
+
+def _input_func_py2_py3():
+    """Find function for reading user input that works on Python 2 and 3.
+
+    See e.g. http://stackoverflow.com/questions/21731043/use-of-input-raw-input-in-python-2-and-3
+    """
+    try:
+        input = raw_input
+    except NameError:
+        import builtins
+        input = builtins.input
+    return input
+
+
+def _user_verify():
+    input = _input_func_py2_py3()
+    if not input("Perform these computations? [y/n] ").lower()[0] == 'y':
+        raise IOError('\n', 'Execution cancelled by user.')
+
+
+def _get_all_objs_of_type(type_, parent):
+    return set([obj for obj in parent.__dict__.values()
+                if isinstance(obj, type_)])
+
+
+class CalcSuite(object):
+    """Generate suite of Calc objects based on inputted specifications."""
+
+    _CORE_SPEC_NAMES = ('projects',
+                        'models',
+                        'runs')
+
+    _AUX_SPEC_NAMES = ('variables',
+                       'date_ranges',
+                       'regions',
+                       'input_time_intervals',
+                       'input_time_datatypes',
+                       'input_time_offsets',
+                       'input_vertical_datatypes',
+                       'output_time_intervals',
+                       'output_time_regional_reductions',
+                       'output_vertical_reductions')
+
+    _NAMES_SUITE_TO_CALC_MAPPING = {
+        'projects': 'proj',
+        'models': 'model',
+        'runs': 'run',
+        'variables': 'var',
+        'date_ranges': 'date_range',
+        'regions': 'region',
+        'input_time_intervals': 'intvl_in',
+        'input_time_datatypes': 'dtype_in_time',
+        'input_time_offsets': 'time_offset',
+        'input_vertical_datatypes': 'dtype_in_vert',
+        'output_time_intervals': 'intvl_out',
+        'output_time_regional_reductions': 'dtype_out_time',
+        'output_vertical_reductions': 'dtype_out_vert',
+    }
+
+    def __init__(self, calc_suite_specs, obj_lib):
+        """Turn all inputs into aospy-ready objects."""
+        self._specs_in = calc_suite_specs
+        self._obj_lib = obj_lib
+
+        self._projects = self._get_projects()
+
+    def _get_projects(self):
+        projs = _get_all_objs_of_type(Proj, self._obj_lib)
+        names = self._specs_in['projects']
+        return set([proj for proj in projs if proj.name in names])
+
+    def _get_regions(self):
+        return [_get_all_objs_of_type(Region, self._obj_lib)]
+
+    def _get_variables(self):
+        if hasattr(self._obj_lib, 'variables'):
+            return self._obj_lib.variables
```

Comment:
I've gone ahead and tried this in my object library; this line does get called, but it causes a problem because this function needs to return an iterable to be used in [line 53](https://github.com/spencerahill/aospy/pull/155/files#diff-e3c273a553369da44ac80bfa80ca1eebR53). 

The obvious change (which I tried) would be to return the result of ```_get_all_objs_of_type(Var, self._obj_lib.variables)```, but this results in ```automate``` attempting to create calculations involving all the variables defined in my object library (ignoring the one I specified in the specifications), prompting my comments below.  Is there a simple way to address this?